src/actions/NekInitAction.C

```diff
@@ -102,7 +102,7 @@ NekInitAction::act()
   }
 
   // setup actions only needed if coupling with MOOSE
-  if (_type == "NekRSProblem")
+  if (_type == "NekRSProblem" || _type == "NekRSSeparateDomainProblem")
```

Comment:
What should happen if the NekRS case doesn't have temperature at all? As in, it only solves for the flow? Is that an error, or do you want to allow it?

Reply:
I want to allow that because I want temperature to be an optional coupling, where as flow is always coupled. If we do have temperature coupling but nekrs doesn't have temperature, than we do want an error. So I will add that error for sure.

Reply:
The reason I ask is because within this if-statement at line 105, Cardinal is going to throw an error if the NekRS problem _doesn't_ have temperature (because we need temperature to exist for `NekRSProblem`). So if we're okay with temperature not existing for `NekRSSeparateDomainProblem`, we should split up this if statement into two parts (one that only applies for `NekRSProblem` that errors if temperautre doesn't exist, and a second that initializes scratch for us - which we want for both problems)